crypto/keys/private.go

```diff
@@ -26,15 +28,24 @@ import (
 	"io/ioutil"
 
 	"github.com/google/trillian"
+	"github.com/google/trillian/crypto/keyspb"
 )
 
+// The size of an RSA key generated by this library, in bits, if not overridden.
+const defaultRsaKeySizeInBits = 2048
+
 // SignerFactory creates signers for Trillian trees.
 // A signers may be created by loading a private key, interfacing with a HSM,
 // or sending network requests to a remote key management service, to give a few
 // examples.
 type SignerFactory interface {
 	// NewSigner returns a signer for the given tree.
+	// It consults Tree.PrivateKey to determine how to retrieve the key.
 	NewSigner(context.Context, *trillian.Tree) (crypto.Signer, error)
+
+	// Generate creates a new asymmetric key pair for a tree according to some parameters.
+	// It returns the tree with updated PrivateKey and PublicKey fields.
```

Comment:
nit: "It returns a new tree with updated ..."

Any reason you chose this route instead of modifying the input tree?

Conceptually, I would prefer it to do neither and instead return `(PrivateKey, error)`; That would mean moving all the key-derivation and Any-handling logic to the Admin server, which seems better as well (it already does key derivation).